framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -479,4 +479,15 @@ namespace cppmicroservices
         }
         return false;
     }
+
+    ServiceReferenceU
+    ServiceReferenceFromService(std::shared_ptr<void> const& s)
+    {
+        auto deleter = std::get_deleter<CustomServiceDeleter>(s);
+        if (!deleter)
+        {
+            throw std::runtime_error("Not a managed shared_ptr");
```

Comment:
Could add test coverage of this by confirming that this throws if you pass in a `shared_ptr` you've created yourself in the test?